File: medical_management_BE/src/main/java/com/nhom2/qnu/service/RoomService.java
package com.nhom2.qnu.service;

import com.nhom2.qnu.model.Room;
import java.util.List;

import org.springframework.stereotype.Service;

@Service
public interface RoomService {

    Room create(Room room);

    Room update(String roomId, Room room);

    void delete(String roomId);

    List<Room> getAll();

}

File: medical_fe/src/services/roomService.js
import api from "../core/api";

const roomService = {

    // ==========================================
    // GET ALL ROOMS
    // ==========================================
    getAll() {

        return api.get("/rooms");
    },

    // ==========================================
    // GET ROOM BY ROOM GROUP
    // ==========================================
    getByRoomGroup(roomGroupId) {

        return api.get(
            `/rooms/by-room-group/${roomGroupId}`
        );
    },

    // ==========================================
    // CREATE ROOM
    // ==========================================
    create(data) {

        return api.post(
            "/rooms",
            data
        );
    },

    // ==========================================
    // UPDATE ROOM
    // ==========================================
    update(roomId, data) {

        return api.put(
            `/rooms/${roomId}`,
            data
        );
    },

    // ==========================================
    // DELETE ROOM
    // ==========================================
    delete(roomId) {

        return api.delete(
            `/rooms/${roomId}`
        );
    }
};

export default roomService;